### DISCUSSION TITLE
debian build

### DISCUSSION THREAD

**Discussion by i30817** [Difegue/LANraragi#958] [General]
My only PC is a 32 bits eeepc from 2008 (901, a first gen atom, can't run 64 bits because of the board).

Needless to say, none of the install methods works for me (homebrew needs 64 bits, docker is too heavy), and I don't have enough space for building on the pc.

So I'm asking for debian (it's a antix 23.1 install, which is based on debian 12) builds, so I can use it as a ebooks epub and comics server. I managed to use nfs to use it as a android music and movies server for vlc even if the movie usecase is... slow.

**Comment by Difegue** [comment]
👋 There's technically nothing to build per se since the entire server is perl scripts, you should be able to get things running with the building from source steps fairly easily.

The instructions are based on debian so you can just install their 32bit packages - - Only parts that might take some space are the cpan modules and npm's hellmachine to grab javascript frontend dependencies 🤔

The frontend stuff can be extracted from another build if needed, so you'd only have to run `install-back` on the host machine for cpan.

**Comment by i30817** [comment]
I don't actually plan to use the web interface, can I just.... Ignore the frontend or will it explode without?

**Comment by Difegue** [comment]
🤔 The server should be fine without it, but you'll need to use the web interface for initial configuration though..

**Comment by i30817** [comment]
I ended up doing a mount point for a /nix directory with the logged in user owner and permissions 755 to a external driver and installed the single user nix package. With the persistent install not the one the doc points to. And using a temp also pointing to the external drive otherwise the 5 gb of downloaded stuff chokes on extracting or building lol.

`TMPDIR=/media/i3/Mordred/tmp/ nix-env -iA nixpkgs.lanraragi`

Then in the script that does the login I used the mount proxy of antix to mount after the drive is mounted (Which also uses udevil, so not on fstab), then run lanraragi (haven't finished installing, let's see if it even works).

Bless Linux and posix being so configurable, pretty sure if I was using windows xp like this computer originally was it'd be hopeless.

**Reply by Difegue** [reply]
I applaud the attempt if nothing else, my atom eeepc is lingering in a drawer and I can't bring myself to throw it out because it has a touch screen...

**Reply by i30817** [reply]
Unfortunately nix recognized that the nix paths is not on root (somehow) which forces compilation instead of binary install, and installing lanraragi is building a whole lot of stuff on a minipc from 2008 now. Been building a day and no where near over. At least it's now on the JavaScript compiler, so maybe just one more day?

I actually managed to make the PC be a webdav server for a games drive with apache dav (in the server) and RSAF (in android). Doesn't work for RetroArch because they haven't implemented SAF for years now, but works for scummvm, dolphin and PPSSPP. I'm also using nfs for VLC music and movie streaming but I wish they had webdav streaming too. For some strange reason, vlc bugs out on nfs filenames with a '?' on them (from a playlist, so no pretending the filenames are something else will work), probably a ill conceived use of a url encoder because the paths are being saved as a 'uri'. The exact reason I changed to nfs from samba was the silly windows limitations on the filenames, so that's pretty disappointing.

**Comment by i30817** [comment]
After 2 days building the nixos "install" failed with "perl5.38.2-FFI-Platypus-2.08.drv: failed with exit code 2"

Fuck me for using a 32bits machine I guess.

**Reply by i30817** [reply]
Hu. It might have been the recently disclosed xs backdoor. I retried it and part of the things that fail is libarchive and I think I noticed xs. Maybe perl is disabling the download of projects the identity\pseudonymn the black hat group or person used.

**Reply by Difegue** [reply]
I believe the compromised version of xz never made it to debian stable, but idk what nix is trying to use there

**Reply by i30817** [reply]
The attacker also contributed to libarchive.


https://boehs.org/node/everything-i-know-about-the-xz-backdoor
